app/models/conversion_host/configurations.rb

```diff
@@ -32,8 +32,7 @@ def enable_queue(params, auth_user = nil)
       resource_type = params[:resource_type]
       resource_id = params[:resource_id]
 
-      resource = resource_type.to_s.downcase.classify.constantize.find(resource_id)
-
+      resource = resource_type.to_s.classify.constantize.find(resource_id)
```

Comment:
@djberg96 could `enable_queue` just take the resource and then this method could split that up into `resource_type` and `resource_id`?

Reply:
It would be called via the rest api. I don't think anything calls it internally (not yet, anyway).

Reply:
Why because the rest api will have the resource_type/id but not the resource?

Reply:
Pretty sure. @jameswnl We'll just have the `resource_id` and `resource_type` right? Or could we have an actual resource object being passed in?

I suppose we could modify it to handle both.

Reply:
I assume the UI will be hitting http endpoint of API and so it's API's decision whether it will make a AR object out of the (json) parameters coming through http request, and then send the AR record to `enable_queue`

Reply:
Ok, I vote to leave it alone for now, and update it later in a separate PR if needed.

Reply:
I would suggest to revert back to the very first version: `resource_type.constantize.find(resource_id)` and leave it to the caller to make the right choice and let it blow if it's wrong. It's not backend's responsibility to fix the class names.

Reply:
@jameswnl that seems unfriendly. Besides, it's just a `.classify` call at this point.

Reply:
by blow up, I mean let it error out, API users are expected to follow specs. 

I am fine with what you have now.

Reply:
Can you add tests?

Reply:
@jameswnl Ok, added a test.

Reply:
@djberg96 I still think this would be a better interface if we pass in the resource from the API code, this type of lookup is done all the time in the API and there are helper methods for it (lookup `resource_search` in `manageiq-api`)

Reply:
@agrare Can you give me an idea of what you're thinking for the endpoint, e.g. what would a curl request look like for what you have in mind?

Reply:
I think that is up to the API guys to decide...I could see posting a create to `/conversion_hosts` and pass the resource_type/id in the body or adding an action like `/vms/1/enable_conversion_host` where the resource is pre-selected

That's not really directly related to this though because either one would be able to do the `resource_search` from the API.

Reply:
@agrare The first case would be covered by https://github.com/ManageIQ/manageiq-api/pull/535 (which is dependent on this PR to pass tests). I'm not sure if the v2v team needs the second case yet, and would like to hold off on that for now. I could create a separate PR for that if needed.

Reply:
@djberg96 I'm still not sure what you're talking about :smile: The API has helper methods to do what you had to do here manually and I think it'd be better to pass a `resource` in here instead of a `resource_type` + `resource_id`

Reply:
We have to accept params anyway, because they're forwarded as args to the `queue_configuration` method. Again, I'd rather not mess with the existing API unless the v2v team comes back with a request to do so.

Reply:
I don't think we'd have to mess with the API, we'd just need to have `data['resource'] = resource_search(data['resource_id'], data['resource_type'])` here https://github.com/ManageIQ/manageiq-api/pull/535/files#diff-2da9489e8f3f5aa77ab1c89559de7604R29

Reply:
It was a little trickier than that, as `resource_search` requires a collection class as a 3rd argument, but I've added that logic, and updated this PR to look for just a `:resource`.

Reply:
Looks like there's an issue with your rebase